lib/update/cluster/phases/init.go

```diff
@@ -491,3 +503,126 @@ func (p *updatePhaseInit) removeConfiguredPackages() error {
 			return nil
 		})
 }
+
+func (r peerCleaner) cleanPeers() error {
+	if err := r.removeInvalidObjectPeerRefs(); err != nil {
+		return trace.Wrap(err)
+	}
+	if err := r.removeLostObjects(); err != nil {
+		return trace.Wrap(err)
+	}
+	if err := r.removeInvalidObjectPeers(); err != nil {
+		return trace.Wrap(err)
+	}
+	return nil
+}
+
+// removeLostObjects removes the object hashes from the state database
+// that only have peers that are no longer in the cluster
+func (r peerCleaner) removeLostObjects() error {
+	objects, err := r.p.GetObjects()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	var errors []error
+	for _, hash := range objects {
+		peers, err := r.p.GetObjectPeers(hash)
+		if err != nil && !trace.IsNotFound(err) {
+			return trace.Wrap(err)
+		}
+		if !r.invalidPeers(peers...) {
+			continue
+		}
+		r.log.WithField("hash", hash).Info("Remove lost object.")
```

Comment:
I'd say "Remove object without peers" and log the peers as well.

Reply:
I opted out of object deletion.